.github/workflows/publish-kn.yml

```diff
@@ -0,0 +1,54 @@
+name: publish-kn
+
+on:
+  workflow_dispatch:
+
+jobs:
+  publish:
+    name: Library Publish
+
+    runs-on: macos-latest
+
+    defaults:
+      run:
+        working-directory: ./wrappers/kn
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Checkout toolchain initializer repository
+        uses: actions/checkout@v4
+        with:
+          repository: ISNing/kn-toolchain-initializer
+          path: wrappers/kn/.kn-toolchain-initializer
+
+      - name: Set up JDK
+        uses: actions/setup-java@v4
+        with:
+          java-version: 17
+          distribution: temurin
+
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v3
+
+      - name: Validate Gradle Wrapper
+        uses: gradle/wrapper-validation-action@v2
+
+      - name: Initialize Kotlin/Native toolchain
+        working-directory: wrappers/kn/.kn-toolchain-initializer
+        run: ./gradlew build -DkotlinVersion=1.9.22
+
+      - name: Export Toolchain properties
+        run: |
+          echo -e "konan.dir=$HOME/.konan/kotlin-native-prebuilt-macos-x86_64-1.9.22" > local.properties
+
+      - name: Publish Library
+        env:
+          ORG_GRADLE_PROJECT_signingKey: ${{ secrets.OSSRH_GPG_SECRET_KEY }}
+          ORG_GRADLE_PROJECT_signingPassword: ${{ secrets.OSSRH_GPG_SECRET_KEY_PASSWORD }}
+          ORG_GRADLE_PROJECT_ossrhUsername: ${{ secrets.OSSRH_USERNAME }}
+          ORG_GRADLE_PROJECT_ossrhPassword: ${{ secrets.OSSRH_PASSWORD }}
+        run: |
+          export PATH="$PATH:$(brew --prefix llvm@15)/bin"
```

Comment:
Looks like you missed this line here, since `brew` is not part of the game anymore.

Reply:
Umm, this is required for building:
https://github.com/zxing-cpp/zxing-cpp/blob/09d300714b965b080d7d133738397b893570b996/wrappers/kn/README.md?plain=1#L52

And the package `llvm@15` is preinstalled by github in action runner images so we don't need to request this package manually:
https://github.com/actions/runner-images/blob/395174881d2f6fd203a41f64f80681cabc81eb25/images/macos/macos-12-Readme.md?plain=1#L16

Reply:
Ah I see now. I thought this would have been related to your earlier `brew` based installation of the kotlin toolchain. And since you removed this `export PATH` line in the `ci.yml` I just thought this was an oversight. But there you switched from macos to ubuntu, which is obviously the reason.

That said: the information you linked to states that llvm 14 is available out of the box, no PATH setting required. So I would assume this should still work without this line. Or is there a reason your plugin insists on llvm-ar-15?

And now I just noticed that there is no `llvm-ar` binary, at least not on ubuntu. So is there a `llvm-ar-15` call hard-coded somewhere?

Reply:
Seems llvm14 installed on macos doesn't have `llvm-ar` tool (at least I don't know where is it, and it's not in `$PATH`) (This also applies for `lld`, although we don't need it here)
And the llvm installed with brew has all utils installed but it needs configuring `$PATH`

And for ubuntu, all the installed packages binaries will bin linked to `/usr/bin` which is already in `$PATH`, so we don't need to set PATH manually.

Reply:
> Seems llvm14 installed on macos doesn't have `llvm-ar` tool

I'm not super surprised since my Ubuntu does not have one either, as I just explained. There are only versioned files like `llvm-ar-15` available. [10 minutes of digging later...] I had only 'versioned' llvm packages installed. Now after `apt install llvm` I have `llvm-ar` available. Weird that the default llvm installation on GH macos would not provide the generic/non-versioned tools. I'm pretty sure that `llvm-ar-14` is available.

Anyway, my idea about this being unnecessary was obviously wrong. Thanks.